python/grass/jupyter/interactivemap.py

```diff
@@ -347,13 +355,91 @@ def add_layer_control(self, **kwargs):
         else:
             self.layer_control_object = self._ipyleaflet.LayersControl(**kwargs)
 
+    def add_query_button(self):
+        """Add custom features like query button and coordinate retrieval"""
+        import ipywidgets as widgets
+
+        # A ToggleButton to activate/deactivate query mode
+        query_toggle_button = widgets.ToggleButton(
+            icon="crosshairs",
+            value=False,
+            tooltip="Click to activate/deactivate query mode",
+            layout=widgets.Layout(width="33px", margin="0px 0px 0px 0px"),
+        )
+
+        # Function to handle toggle state change
+        def on_toggle_change(change):
+            self.query_mode = change.new
+            # Change cursor style based on query mode
+            self.map.default_style = {
+                "cursor": "crosshair" if self.query_mode else "default"
+            }
+
+        query_toggle_button.observe(on_toggle_change, names="value")
+
+        # Add the toggle button to the map
+        query_control = self._ipyleaflet.WidgetControl(
+            widget=query_toggle_button, position="topright"
+        )
+        self.map.add_control(query_control)
+
+        # Function to handle map click for querying
+        def handle_interaction(**kwargs):
+            if not self.query_mode:
+                return
+            if self.query_mode and kwargs.get("type") == "click":
+                lonlat = kwargs.get("coordinates")
+                reprojected_coordinates = reproject_latlon((lonlat[0], lonlat[1]))
+                raster_output = query_raster(
+                    (reprojected_coordinates[0], reprojected_coordinates[1]),
+                    self.raster_name,
+                )
+                vector_output = query_vector(
+                    (reprojected_coordinates[0], reprojected_coordinates[1]),
+                    self.vector_name,
+                )
+                message = widgets.HTML()
+                message.value = raster_output + vector_output
+
+                # Create a scrollable popup
+                scrollable_container = widgets.HTML(
+                    value=(
+                        "<div style='max-height: 300px; max-width: 300px; "
+                        "overflow-y: auto; overflow-x: auto;'>"
+                        f"{message.value}"
+                        "</div>"
+                    )
+                )
+
+                # Popup with a given location on the map:
+                popup = self._ipyleaflet.Popup(
+                    location=lonlat,
+                    child=scrollable_container,
+                    close_button=False,
+                    auto_close=False,
+                    close_on_escape_key=False,
+                )
+                self.map.add(popup)
+
+            def on_toggle_popup_change(change):
+                if not change.new:
+                    # Find and remove all popups on the map
+                    for item in list(self.map.layers):
```

Comment:
This helps with the flickering:
```suggestion
                    for item in reversed(list(self.map.layers)):
```

Reply:
Hi Anna,
I can't see any flickering issue in the popup. Is it still there on your side?

Reply:
When you are removing the popups, it flickers. with this change it is much better.